src/GenFacades/GenFacades.Core/GenFacades.cs

```diff
@@ -433,7 +453,27 @@ public Assembly GenerateFacade(IAssembly contractAssembly, IAssemblyReference se
                         continue;
                     }
 
+                    if (buildPartialReferenceFacade)
+                    {
+                        // honor preferSeedType for keeping contract type
+                        string preferredSeedAssembly;
+                        bool keepType = _seedTypePreferences.TryGetValue(docId, out preferredSeedAssembly) &&
+                            contractAssemblyName.Equals(preferredSeedAssembly, StringComparison.OrdinalIgnoreCase);
+
+                        if (!keepType)
+                        {
+                            assembly.AllTypes.Remove(existingDocIds[docId]);
+                        }
+                    }
+
                     AddTypeForward(assembly, seedType);
+                    addedForward = true;
+                }
+
+                if (buildPartialReferenceFacade &&  !addedForward)
+                {
+                    Trace.TraceError("Did not find any types in any of the seed assemblies.");
```

Comment:
I assume in this case it is either an error or you should no longer set IsPartialFacade.